code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -429,36 +281,342 @@
+						return FALSE
+				else
+					M.unEquip(I)
+				I.forceMove(src)
+			else
+				I.forceMove(src)
+
+			item_quants[I.name] += 1
+			return TRUE
 	return FALSE
 
+/**
+  * Tries to shoot a random at a nearby living mob.
+  */
 /obj/machinery/smartfridge/proc/throw_item()
-	var/obj/throw_item = null
-	var/mob/living/target = locate() in view(7,src)
+	var/obj/item/throw_item = null
+	var/mob/living/target = locate() in view(7, src)
 	if(!target)
-		return 0
+		return FALSE
 
 	for(var/O in item_quants)
 		if(item_quants[O] <= 0) //Try to use a record that actually has something to dump.
 			continue
-
 		item_quants[O]--
-		for(var/obj/T in contents)
-			if(T.name == O)
-				T.forceMove(loc)
-				throw_item = T
+		for(var/obj/I in contents)
+			if(I.name == O)
+				I.forceMove(loc)
+				throw_item = I
 				update_icon()
 				break
-		break
 	if(!throw_item)
-		return 0
-	spawn(0)
-		throw_item.throw_at(target,16,3,src)
+		return FALSE
+
+	INVOKE_ASYNC(throw_item, /atom/movable.proc/throw_at, target, 16, 3, src)
 	visible_message("<span class='warning'>[src] launches [throw_item.name] at [target.name]!</span>")
-	return 1
+	return TRUE
+
+/**
+  * Returns whether the smart fridge can accept the given item.
+  *
+  * By default checks if the item is in [the typecache][/obj/machinery/smartfridge/var/accepted_items_typecache].
+  * Arguments:
+  * * O - The item to check.
+  */
+/obj/machinery/smartfridge/proc/accept_check(obj/item/O)
+	return is_type_in_typecache(O, accepted_items_typecache)
+
+/**
+  * # Secure Fridge
+  *
+  * Secure variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  * Can be emagged and EMP'd to short the lock.
+  */
+/obj/machinery/smartfridge/secure
+	is_secure = TRUE
+
+/obj/machinery/smartfridge/secure/emag_act(mob/user)
+	emagged = TRUE
+	locked = SMART_FRIDGE_LOCK_SHORTED
+	to_chat(user, "<span class='notice'>You short out the product lock on \the [src].</span>")
+
+/obj/machinery/smartfridge/secure/emp_act(severity)
+	if(!emagged && locked != SMART_FRIDGE_LOCK_SHORTED && prob(40 / severity))
+		playsound(loc, 'sound/effects/sparks4.ogg', 60, TRUE)
+		emagged = TRUE
+		locked = SMART_FRIDGE_LOCK_SHORTED
+
+/obj/machinery/smartfridge/secure/Topic(href, href_list)
+	if(stat & (BROKEN|NOPOWER))
+		return FALSE
+
+	if(href_list["vend"] && (usr.contents.Find(src) || Adjacent(usr)))
+		if(!emagged && locked != SMART_FRIDGE_LOCK_SHORTED && scan_id && !allowed(usr))
+			to_chat(usr, "<span class='warning'>Access denied.</span>")
+			SSnanoui.update_uis(src)
+			return FALSE
+
+	return ..()
 
-// ----------------------------
-//  Drying Rack 'smartfridge'
-// ----------------------------
+/**
+  * # MegaSeed Servitor
+  *
+  * Seeds variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  */
+/obj/machinery/smartfridge/seeds
+	name = "\improper MegaSeed Servitor"
+	desc = "When you need seeds fast!"
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "seeds"
+
```

Comment:
This is the exact same name as the vending machine. This is not only confusing, but made me double check that botany actually used a proper vending machine.

Reply:
Wasn't aware of that, I renamed it to `Seed Storage` (bland, I know)